src/main/java/com/ait/lienzo/client/core/shape/TextTruncateWrapper.java

```diff
@@ -0,0 +1,239 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ait.lienzo.client.core.shape;
+
+import java.util.ArrayList;
+
+import com.ait.lienzo.client.core.Context2D;
+import com.ait.lienzo.client.core.types.BoundingBox;
+
+/**
+ * ITextWrapper implementation that truncates text and appends "..." if there is no space left.
+ */
+@SuppressWarnings("Duplicates")
+public class TextTruncateWrapper extends TextNoWrap {
+
+    private BoundingBox wrapBoundaries;
+    private final double Y_OFFSET = 0.8;
+
+    public TextTruncateWrapper(final Text text,
+                               final BoundingBox wrapBoundaries) {
+        super(text);
+        setWrapBoundaries(wrapBoundaries);
+    }
+
+    public BoundingBox getWrapBoundaries() {
+        return wrapBoundaries;
+    }
+
+    public TextTruncateWrapper setWrapBoundaries(final BoundingBox boundaries) {
+        wrapBoundaries = boundaries;
+        return this;
+    }
+
+    @Override
+    public BoundingBox getBoundingBox() {
+        final double[] boundaries = calculateWrapBoundaries();
+        return new BoundingBox().addX(0).addX(wrapBoundaries.getWidth()).addY(0).addY(boundaries[1]);
+    }
+
+    private double getWrapBoundariesWidth() {
+        final double margin = 4 * getBoundingBoxForString(" ").getWidth();
+        return this.wrapBoundaries.getWidth() - margin;
+    }
+
+    private double[] calculateWrapBoundaries() {
+        final String[] words = textSupplier.get().split("\\s");
+        if (words.length < 1) {
+            return new double[]{wrapBoundaries.getX(), wrapBoundaries.getY()};
+        }
+
+        final double wrapWidth = getWrapBoundariesWidth();
+        final String firstWord = words[0];
+        double width = getBoundingBoxForString(firstWord).getWidth();
+        final StringBuilder nextLine = new StringBuilder(firstWord);
+        int numOfLines = 1;
+        for (int i = 1; i < words.length; i++) {
```

Comment:
This seems to work out how many lines the text needs to be split into with words fitting the `wrapWidth`..

Reply:
Yes, but is not very precise at this point, is a estimative for the worst case.